Tools/i18n/pygettext.py

```diff
@@ -343,6 +344,34 @@ def __waiting(self, ttype, tstring, lineno):
                 return
         if ttype == tokenize.NAME and tstring in opts.keywords:
             self.__state = self.__keywordseen
+            return
+        if ttype == tokenize.STRING:
+            maybe_fstring = ast.parse(tstring, mode='eval').body
+            if not isinstance(maybe_fstring, ast.JoinedStr):
+                return
+            for value in filter(lambda node: isinstance(node, ast.FormattedValue),
+                                maybe_fstring.values):
+                for call in filter(lambda node: isinstance(node, ast.Call),
+                                   ast.walk(value)):
+                    func = call.func
+                    if isinstance(func, ast.Name):
+                        func_name = func.id
+                    elif isinstance(func, ast.Attribute):
+                        func_name = func.attr
+                    else:
+                        continue
+
+                    if func_name not in opts.keywords:
+                        continue
+                    if len(call.args) != 1 or call.keywords:
+                        # what warning should I print when this happens?
+                        continue
+                    arg = call.args[0]
+                    if not isinstance(arg, ast.Constant):
+                        # what warning should I print when this happens?
+                        break
```

Comment:
Is this `break` intentional?

Reply:
I wrote this a while ago, but it seems unintentional to me now and I probably meant to put `continue` here instead.

As an example, with `break` `pygettext` completely ignores "bar" in code like this:
```
f"{_(f'foo') + _('bar')}"
```
which was definitely not my intention.

I've just pushed an update replacing `break` with `continue`, thanks :)